zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -312,8 +312,8 @@ public void parseProperties(Properties zkProp) throws IOException, ConfigExcepti
                 connectToLearnerMasterLimit = Integer.parseInt(value);
             } else if (key.equals("electionAlg")) {
                 electionAlg = Integer.parseInt(value);
-                if (electionAlg != 1 && electionAlg != 2 && electionAlg != 3) {
```

Comment:
perhaps a better location for the error..

Reply:
Didn't make any changes in this part because default value of electionAlg is 3 and if other than 3 is used then configException is thrown instead made changes in QuorumPeer.createElectionAlgorithm